src/Bounds.cpp

```diff
@@ -552,7 +552,16 @@ class Bounds : public IRVisitor {
             int min_sign = static_sign(b.min);
             int max_sign = static_sign(b.max);
             if (min_sign != max_sign || min_sign == 0 || max_sign == 0) {
-                interval = Interval::everything();
+                if (op->type.is_float()) {
+                    interval = Interval::everything();
+                } else {
```

Comment:
Does this need to ensure we're dealing with no_overflow types?

You may already be accounting for this, but I think `INT_MIN/-1 > INT_MAX` for signed types.